src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -60,18 +63,68 @@ public class ScriptHost : JobHost
         private bool _shutdownScheduled;
         private ILogger _startupLogger;
         private FileWatcherEventSource _fileEventSource;
-        private IDisposable _fileEventsSubscription;
+        private IList<IDisposable> _subscriptions = new List<IDisposable>();
         private ProxyClientExecutor _proxyClient;
+        private IFunctionDispatcher _functionDispatcher;
         private ILoggerFactory _loggerFactory;
+        private IProcessRegistry _processRegistry = new EmptyProcessRegistry();
+
+        // Specify the "builtin binding types". These are types that are directly accesible without needing an explicit load gesture.
+        // This is the set of bindings we shipped prior to binding extensibility.
+        // Map from BindingType to the Assembly Qualified Type name for its IExtensionConfigProvider object.
+        // TODO: Re-add built in providers
+        private static IReadOnlyDictionary<string, string> _builtinBindingTypes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            //{ "bot", "Microsoft.Azure.WebJobs.Extensions.BotFramework.Config.BotFrameworkConfiguration, Microsoft.Azure.WebJobs.Extensions.BotFramework" },
+            //{ "sendgrid", "Microsoft.Azure.WebJobs.Extensions.SendGrid.SendGridConfiguration, Microsoft.Azure.WebJobs.Extensions.SendGrid" },
+            //{ "eventGridTrigger", "Microsoft.Azure.WebJobs.Extensions.EventGrid.EventGridExtensionConfig, Microsoft.Azure.WebJobs.Extensions.EventGrid" }
+        };
+
+        private static IReadOnlyDictionary<string, string> _builtinScriptBindingTypes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
See my other comment - I had factored all this stuff out into an ExtensionLoader type

Reply:
See my reply :)